src/test/html/html-element-reference-scanner_test.ts

```diff
@@ -0,0 +1,42 @@
+import {assert} from 'chai';
+import {HtmlVisitor} from '../../html/html-document';
+import {HtmlParser} from '../../html/html-parser';
+import {HtmlElementReferenceScanner} from '../../html/html-element-reference-scanner';
+import {ScannedElementReference} from '../../model/element-reference';
+
+suite('HtmlElementReferenceScanner', () => {
+
+  suite('scan()', () => {
+    let scanner: HtmlElementReferenceScanner;
+
+    setup(() => {
+      scanner = new HtmlElementReferenceScanner();
+    });
+
+    test('finds custom element references', async() => {
+      let contents = `<html><body>
+          <div>Foo</div>
+          <x-foo></x-foo>
+          <div>
+            <x-bar></x-bar>
+          </div>
+          <h1>Bar</h1>
+        </body></html>`;
+      const document = new HtmlParser().parse(contents, 'test-document.html');
+      let visit = async(visitor: HtmlVisitor) => document.visit([visitor]);
+
+      const features = await scanner.scan(document, visit);
+      assert.equal(features.length, 2);
+      assert.instanceOf(features[0], ScannedElementReference);
+      assert.instanceOf(features[1], ScannedElementReference);
+
+      const feature0 = <ScannedElementReference>features[0];
+      assert.equal(feature0.tagName, 'x-foo');
```

Comment:
I like using `assert.deepEquals` more than a lot of little assertions, because you get way better error messages, with diffs.

So e.g.

```typescript
assert.deepEquals(features.map(f => f.tagName), ['x-foo', 'x-bar']);
```